bot/cogs/tags.py

```diff
@@ -86,11 +86,62 @@ def _get_suggestions(self, tag_name: str, thresholds: Optional[List[int]] = None
             return self._get_suggestions(tag_name)
         return found
 
+    async def _get_tags_via_content(self, check: Callable[[Iterable], bool], keywords: str) -> Optional[Embed]:
+        """
+        Search for tags via contents.
+
+        `predicate` will be either any or all, or a custom callable to search. Must return a bool.
+        """
+        await self._get_tags()
+
+        keywords_processed: Tuple[str] = tuple(query.strip().casefold() for query in keywords.split(',') if query)
+        keywords_processed = keywords_processed or (keywords,)
+        founds: list = [
+            tag
+            for tag in self._cache.values()
+            if check(query in tag['embed']['description'] for query in keywords_processed)
+        ]
+
+        if not founds:
+            return None
+        elif len(founds) == 1:
+            return Embed().from_dict(founds[0]['embed'])
+        else:
+            is_plural: bool = len(keywords_processed) > 1 or any(kw.count(' ') for kw in keywords_processed)
```

Comment:
wait, can you explain the `or any(kw.count(' ') for kw in keywords_processed)` part of this for me? 

Also, this line is dense, too, and while I don't mind the type annotation here as much, I think the excellent variable name makes it intuitively clear that this is a boolean.

Reply:
Since I am allowing searching with multiple keywords, separated by comma at the same time, simply counting for `' '` will not work, for example if use search for `code,block` - there will be no space at all in the `keywords`

After thinking about this more, it is definitely over kill with my original intention to check if any `separated query` has any space in it, so the line can be reduced to
```
is_plural = len(keywords_processed) > 1 or keywords.strip().count(' ')
```